File: src/cujae/inf/citi/om/problem/input/Depot.java
package cujae.inf.citi.om.problem.input;

import java.util.ArrayList;

public class Depot {
	
	private int idDepot;
	private Location locationDepot;
	private ArrayList<Fleet> fleetDepot;

	public Depot() {
		super();
		fleetDepot = new ArrayList<Fleet>();
	}
	 
	public Depot(int idDepot, Location locationDepot,
			ArrayList<Fleet> fleetDepot) {
		super();
		this.idDepot = idDepot;
		this.locationDepot = locationDepot;
		this.fleetDepot = fleetDepot;
	}

	public int getIDDepot() {
		return idDepot;
	}

	public void setIDDepot(int idDepot) {
		this.idDepot = idDepot;
	}

	public Location getLocationDepot() {
		return locationDepot;
	}

	public void setLocationDepot(Location locationDepot) {
		this.locationDepot = locationDepot;
	}

	public ArrayList<Fleet> getFleetDepot() {
		return fleetDepot;
	}

	public void setFleetDepot(ArrayList<Fleet> fleetDepot) {
		this.fleetDepot = fleetDepot;
	}
}

File: BHAVRP_Python/cujae/inf/citi/om/problem/input/depot.py
from typing import List
from fleet import Fleet
from location import Location

class Depot:
    # Constructor para la clase Depot.
    def __init__(
        self, 
        id_depot: int, 
        location_depot: Location = None, 
        fleet_depot: List[Fleet] = None
    ):
        # Si no se pasa una ubicación, se inicializa con coordenadas por defecto
        if location_depot is None:
            location_depot = Location(0.0, 0.0)
        if fleet_depot is None:
            fleet_depot = []
        
        self.id_depot = id_depot
        self.location_depot = location_depot
        self.fleet_depot = fleet_depot
    
    def get_id_depot(self) -> int:
        return self.id_depot

    def set_id_depot(self, value: int):
        self.id_depot = value

    def get_location_depot(self) -> Location:
        return self.location_depot

    def set_location_depot(self, value: Location):
        self.location_depot = value

    def get_fleet_depot(self) -> List[Fleet]:
        return self.fleet_depot

    def set_fleet_depot(self, value: List[Fleet]):
        self.fleet_depot = value